intermediate/challenge-1/app/api/api_v1/endpoints/login.py

```diff
@@ -0,0 +1,35 @@
+from datetime import timedelta
+from typing import Any
+
+from fastapi import APIRouter, Depends, HTTPException
+from fastapi.security import OAuth2PasswordRequestForm
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from api import deps
+from core import security
+from core.config import settings
+
+router = APIRouter()
+
+
+@router.post("/access-token", response_model=schemas.Token)
+def login_access_token(
+    db: Session = Depends(deps.get_db), form_data: OAuth2PasswordRequestForm = Depends()
+) -> Any:
+    """
+    OAuth2 compatible token login, get an access token for future requests
+    """
+    user = crud.user.authenticate(
+        db, email=form_data.username, password=form_data.password
+    )
+    if not user:
+        raise HTTPException(status_code=400, detail="Incorrect email or password")
+    access_token_expires = timedelta(minutes=settings.ACCESS_TOKEN_EXPIRE_MINUTES)

```

Comment:
I believe you want to import from the `Settings` class, not `settings` module; fixing this could allow you to remove line 62 of config.py, which appears to be not needed or inappropriate.

Due to this apparently, I am unable to run the application, docker-compose returns the below:
```app_1       |   File "/usr/src/app/core/security.py", line 5, in <module>
app_1       |     from core.config import settings
app_1       |   File "/usr/src/app/core/config.py", line 7, in <module>
app_1       |     class Settings(BaseSettings):
app_1       |   File "/usr/src/app/core/config.py", line 35, in Settings
app_1       |     [str] = None
app_1       | TypeError: cannot unpack non-iterable NoneType object
```

While if I try running `main` from IDE, I am getting:
```
File "/Users/abdelrahmansinno/IdeaProjects/temp/amaze-us/intermediate/challenge-1/app/crud/crud_user.py", line 5, in <module>
    from core.security import get_password_hash, verify_password
  File "/Users/abdelrahmansinno/IdeaProjects/temp/amaze-us/intermediate/challenge-1/app/core/security.py", line 5, in <module>
    from core.config import settings
ImportError: cannot import name 'settings' from 'core.config' 
```

Reply:
I don't know why docker-compose returns this error but I removed line 62 in config.py and imported the "settings" module and it worked fine.
for the case of the IDE, I think you should mark the app folder as the source root